tests/z_channels_test.c

```diff
@@ -0,0 +1,118 @@
+//
+// Copyright (c) 2024 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+#include <assert.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "zenoh-pico/api/handlers.h"
+
+#undef NDEBUG
+#include <assert.h>
+
+#define SEND(channel, v)                           \
+    do {                                           \
+        z_sample_t sample;                         \
+        sample.payload.start = (const uint8_t *)v; \
+        sample.payload.len = strlen(v);            \
+        sample.keyexpr = _z_rname("key");          \
+        z_call(channel.send, &sample);             \
+    } while (0);
```

Comment:
Just a curiosity. Are you using here ` do { ... } while (0)` as a way to implement scoping `{ ... }`?

Reply:
I explained the rationale: https://github.com/eclipse-zenoh/zenoh-pico/pull/307#pullrequestreview-1797921122